src/eradiate_plugins/bsdfs/rtls.cpp

```diff
@@ -0,0 +1,322 @@
+        UnpolarizedSpectrum cos_t =
+            (m_h / m_b) * dr::sqrt(dr::sqr(D) + dr::sqr(tan_sin_prod)) /
+            sec_theta_sum;
+
+        // Clip cos(t) values outside of [-1; 1]
+        cos_t = dr::minimum(cos_t, 1.f);
+        cos_t = dr::maximum(cos_t, -1.f);
+        Log(Trace, "cos_t: %s", cos_t);
+
+        const UnpolarizedSpectrum t     = dr::acos(cos_t);
+        const UnpolarizedSpectrum sin_t = dr::sin(t);
+
+        return (1.f / dr::Pi<Float>) *(t - sin_t * cos_t) * sec_theta_sum;
+    }
+
+    inline const UnpolarizedSpectrum
+    eval_K_geo(const Float &cos_theta_i, const Float &cos_theta_o,
+               const Float &tan_theta_i, const Float &tan_theta_o,
+               const Float &cos_d_phi, const Float &sin_d_phi,
+               const Float &cos_psi) const {
+        const Float sec_theta_i = 1.f / cos_theta_i;
+        const Float sec_theta_o = 1.f / cos_theta_o;
+
+        const Float sec_theta_sum = sec_theta_i + sec_theta_o;
+
+        const UnpolarizedSpectrum O = eval_O(
+            tan_theta_i, tan_theta_o, sec_theta_sum, cos_d_phi, sin_d_phi);
+
+        Log(Trace, "O: %s", O);
+
+        return O - sec_theta_sum +
+               0.5 * (1.f + cos_psi) * sec_theta_i * sec_theta_o;
+    }
+
+    const UnpolarizedSpectrum eval_rtls(const SurfaceInteraction3f &si,
+                                        const Vector3f &wo, Mask active) const {
+
+        const UnpolarizedSpectrum f_iso = m_f_iso->eval(si, active);
+        const UnpolarizedSpectrum f_vol = m_f_vol->eval(si, active);
+        const UnpolarizedSpectrum f_geo = m_f_geo->eval(si, active);
+
+        auto [sin_phi_i, cos_phi_i] = Frame3f::sincos_phi(si.wi);
+        auto [sin_phi_o, cos_phi_o] = Frame3f::sincos_phi(wo);
+
+        const Float sin_theta_i = Frame3f::sin_theta(si.wi);
+        const Float cos_theta_i = Frame3f::cos_theta(si.wi);
+        const Float tan_theta_i = Frame3f::tan_theta(si.wi);
+        const Float sin_theta_o = Frame3f::sin_theta(wo);
+        const Float cos_theta_o = Frame3f::cos_theta(wo);
+        const Float tan_theta_o = Frame3f::tan_theta(wo);
+
+        Float cos_d_phi = cos_phi_i * cos_phi_o + sin_phi_i * sin_phi_o;
+        Float sin_d_phi = sin_phi_i * cos_phi_o - cos_phi_i * sin_phi_o;
+
+        std::ostringstream oss;
+        oss << "Geometrical setup" << std::endl
+            << "  phi_i: " << dr::asin(sin_phi_i) << std::endl
+            << "    sin(phi_i): " << sin_phi_i << std::endl
+            << "    cos(phi_i): " << cos_phi_i << std::endl
+            << "  phi_o: " << dr::asin(sin_phi_o) << std::endl
+            << "    sin(phi_o): " << sin_phi_o << std::endl
+            << "    cos(phi_o): " << cos_phi_o << std::endl
+            << "  d_phi: " << dr::asin(sin_d_phi) << std::endl
+            << "    sin(d_phi): " << sin_d_phi << std::endl
+            << "    cos(d_phi): " << cos_d_phi << std::endl;
+        Log(Trace, oss.str().c_str());
```

Comment:
I think you should remove this: these lines will be executed even if the logs are not displayed.